examples/demo/src/reviews/ReviewListMobile.tsx

```diff
@@ -28,12 +29,22 @@ const useStyles = makeStyles({
     },
 });
 
-const ReviewMobileList = ({ basePath, data, ids, loading, total }) => {
+const ReviewListMobile: FC<DatagridProps<Review>> = ({
+    basePath,
+    data,
+    ids,
+    loading,
+    total,
+}) => {
     const classes = useStyles();
-    return (
-        (loading || total > 0) && (
-            <List className={classes.root}>
-                {ids.map(id => (
+
+    return loading || Number(total) > 0 ? (
+        <List className={classes.root}>
+            {ids.map(id => {
+                const item = data[id];
+                if (!item) return null;
```

Comment:
for typescript, `data` is optional so it might be undefined and there is no typesafe guarantee it includes all `ids` which is also might be undefined.